tests/math_utils.test.ts

```diff
@@ -85,3 +90,128 @@ describe("remap", () => {
     expect(remap(2, 0, 1, 0, 10, false)).to.equal(20);
   });
 });
+
+describe("getFrameSizeInScreenPx", () => {
+  it("returns canvas size when frame is the same aspect ratio", () => {
+    const frameSizePx = getFrameSizeInScreenPx([100, 100], [1, 1], 1);
+    expect(frameSizePx).to.deep.equal([100, 100]);
+  });
+
+  it("does not change returned size if frame resolution is higher", () => {
+    const frameResolutions: [number, number][] = [
+      [0.1, 0.1],
+      [1, 1],
+      [10, 10],
+      [100, 100],
+    ];
+    for (const frameResolution of frameResolutions) {
+      const frameSizePx = getFrameSizeInScreenPx([100, 100], frameResolution, 1);
+      expect(frameSizePx).to.deep.equal([100, 100]);
+    }
+  });
+
+  it("constrains by height when frame aspect ratio is taller than canvas", () => {
+    const canvasSizePx: [number, number] = [100, 100];
+    const frameResolution: [number, number] = [0.5, 1];
+    const frameZoom = 1;
+    const frameSizePx = getFrameSizeInScreenPx(canvasSizePx, frameResolution, frameZoom);
+    expect(frameSizePx).to.deep.equal([50, 100]);
+  });
+
+  it("constrains by width when frame aspect ratio is wider than canvas", () => {
+    const canvasSizePx: [number, number] = [100, 100];
+    const frameResolution: [number, number] = [1, 0.5];
+    const frameZoom = 1;
+    const frameSizePx = getFrameSizeInScreenPx(canvasSizePx, frameResolution, frameZoom);
+    expect(frameSizePx).to.deep.equal([100, 50]);
+  });
+
+  it("scales frame dimensions with zoom", () => {
+    const canvasSizePx: [number, number] = [100, 100];
+    const frameResolution: [number, number] = [1, 0.5];
+    const frameZoom = 2;
+    const frameSizePx = getFrameSizeInScreenPx(canvasSizePx, frameResolution, frameZoom);
+    expect(frameSizePx).to.deep.equal([200, 100]);
+  });
+});
+
+describe("convertCanvasOffsetPxToFrameCoords", () => {
+  const makeConversionTester = (props: {
+    frameSizeScreenPx: [number, number];
+    canvasSizePx: [number, number];
+    canvasPanPx: [number, number];
+  }) => {
+    return (canvasOffsetPx: [number, number], expected: [number, number]) => {
+      const frameCoords = convertCanvasOffsetPxToFrameCoords(
+        props.canvasSizePx,
+        props.frameSizeScreenPx,
+        canvasOffsetPx,
+        props.canvasPanPx
+      );
+      expect(frameCoords[0]).equals(expected[0]);
+      expect(frameCoords[1]).equals(expected[1]);
+    };
+  };
+
+  it("maps corners correctly on unscaled frames", () => {
```

Comment:
I don't really like the word "correctly" in tests. it's pretty meaningless. What does "correctly" mean in this context (and what would incorrectly be?)